test/jdk/java/awt/Frame/MinimizeUndecoratedTest.java

```diff
@@ -0,0 +1,139 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.Toolkit;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import javax.imageio.ImageIO;
+
+import jtreg.SkippedException;
+
+/*
+ * @test
+ * @key headful
+ * @bug 6251941
+ * @summary Undecorated frames should be minimizable.
+ * @library /test/lib
+ * @build jtreg.SkippedException
+ * @run main MinimizeUndecoratedTest
+ */
+
+public class MinimizeUndecoratedTest {
+    private static final int SIZE = 300;
+    private static final CountDownLatch isMinimized = new CountDownLatch(1);
+
+    private static Frame frame;
+    private static Robot robot;
+
+    private static volatile Point frameLoc;
+
+    public static void main(String[] args) throws Exception {
+        if (!Toolkit.getDefaultToolkit()
+                    .isFrameStateSupported(Frame.ICONIFIED)) {
+            throw new SkippedException("Test is not applicable as"
+                    + " the Window manager does not support MINIMIZATION");
+        }
+
+        try {
+            robot = new Robot();
+            EventQueue.invokeAndWait(MinimizeUndecoratedTest::createUI);
+            robot.waitForIdle();
+            robot.delay(1000);
+
+            EventQueue.invokeAndWait(() -> frameLoc = frame.getLocationOnScreen());
+
+            Color beforeColor = robot.getPixelColor(frameLoc.x + SIZE / 2,
+                                                    frameLoc.y + SIZE / 2);
+
+            EventQueue.invokeAndWait(() -> frame.setExtendedState(Frame.ICONIFIED));
+            robot.waitForIdle();
+            robot.delay(500);
+
+            if (isMinimized.await(8, TimeUnit.SECONDS)) {
+                EventQueue.invokeAndWait(() -> System.out.println("Frame state: "
+                                               + frame.getExtendedState()));
+                Color afterColor = robot.getPixelColor(frameLoc.x + SIZE / 2,
+                                                       frameLoc.y + SIZE / 2);
+
+                if (beforeColor.equals(afterColor)) {
+                    saveScreenCapture();
+                    throw new RuntimeException("Color before & after minimization : "
+                            + beforeColor + " vs " + afterColor + "\n"
+                            + "Test Failed !! Frame not minimized.");
+                }
+            } else {
+                throw new RuntimeException("Window iconified event not received.");
+            }
```

Comment:
I'd invert the condition of the `if (!isMinimized())`, then everything inside the body of the `if`-block can be unindented.

Reply:
Agreed. Updated.